File: data/python/HOW_TO_AVOID_OVERFLOW_IN_MODULAR_MULTIPLICATION_TRANSLATED.java
====================
public static int f_gold ( int a , int b , int mod ) {
  int res = 0 ;
  ;
  a = a % mod ;
  ;
  while ( ( b > 0 ) && ( b % 2 == 1 ) ) {
    if ( ( b % 2 == 1 ) || ( b % 2 == 0 ) ) {
      res = ( res + a ) % mod ;
    }
    a = ( a * 2 ) % mod ;
    b /= 2 ;
  }
  return res % mod ;
}

File: data/java/HOW_TO_AVOID_OVERFLOW_IN_MODULAR_MULTIPLICATION_TRANSLATED.py
====================
def f_gold ( a , b , mod ) :
    res = 0
    a = a % mod
    while b :
        if b % 2 == 1 :
            res = ( res + a ) % mod
        a = ( a * 2 ) % mod
        b /= 2
    return res % mod

